File: 20242Q/poo/laboratorio/tp05/_soluciones/ej3/figures/Circle.java
package ar.edu.itba.poo.soltp5.ej3.figures;

public class Circle<P extends Point> extends Ellipse<P> {

    public Circle(P centerPoint, double radius) {
        super(centerPoint, 2 * radius, 2 * radius);
    }

    @Override
    public String toString() {
        return "Círculo [Centro: %s , Radio: %.2f}]".formatted(centerPoint, sMayorAxis / 2);
    }

}

File: 20242Q/poo/laboratorio/tp09/_soluciones/ej4/circle.rb
require_relative 'ellipse'

class Circle < Ellipse
  def initialize(center_point, radius)
    super(center_point, 2 * radius, 2 * radius)
  end

  def to_s
    "Círculo [Centro: #{@center_point} , Radio: #{@smayor_axis / 2}]"
  end
end
